precompile/allowlist/test_allowlist.go

```diff
@@ -270,3 +270,22 @@ func RunPrecompileWithAllowListTests(t *testing.T, module modules.Module, newSta
 		})
 	}
 }
+
+func BenchPrecompileWithAllowList(b *testing.B, module modules.Module, newStateDB func(t testing.TB) contract.StateDB, contractTests map[string]testutils.PrecompileTest) {
+	b.Helper()
+
+	tests := AllowListTests(module)
+	// Add the contract specific tests to the map of tests to run.
+	for name, test := range contractTests {
+		if _, exists := tests[name]; exists {
+			b.Fatalf("duplicate bench name: %s", name)
+		}
+		tests[name] = test
+	}
+
+	for name, test := range tests {
```

Comment:
seems we're running tests as our benchmark cases. Do you think it would be better if we define benchmark cases in each precompile and let it run them instead of unit tests?

Reply:
The intention is to make each case portable to be run as both a unit test and benchmark

Reply:
hmm but I don't think some unit tests (those are testing failing cases like sufficient gas, invalid input etc) will be helpful for benchmarks.

Reply:
They should still be valid benchmarks and I actually really like the fact that we can easily define a case that is portable for both a unit test and benchmark.

Any valid test case will still be a valid benchmark, they may just produce a much higher mgas/sec if they are running into an insufficient gas or invalid input since they will consume gas without performing the full operation.

Reply:
I was more thinking of simple success cases for each function (readRole, setRole etc). I.e `setRole` case for assigning roles to a predetermined size of addresses in each iteration (without reverting to original state). E.g `BenchmarkSetRole1000`:
```
        size := 1000
	for i := 0; i < b.N; i++ {
		state := newStateDB()
                for k := 0; k < size; k++{
                    address := randAddress(k)
                    input := packSetRole(address)
		    ..., err := contract.Run(state, caller, contractAddress, input, gas, readonly)
		    req.NoErr(err)
		}
	}
```